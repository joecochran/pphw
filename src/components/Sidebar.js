import React from 'react';
import Draggable from './Draggable';

function Sidebar(props) {
  return (
    <div className="c-sidebar">
      <Draggable name="Add lane" handleDrag={props.handleLaneDrag} />
      <Draggable name="Add bar" handleDrag={() => {return false}} />
    </div>
  )
}

export default Sidebar;
