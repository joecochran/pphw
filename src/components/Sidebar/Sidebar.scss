.c-sidebar {
  width: 185px;
  flex-shrink: 0;
  background-color: #FFF;
  padding: 16px;
  border-left: 1px solid #C1C4C7;
  position: relative;

  &__onboarding {
    position: absolute;
    left: -484px;
    top: -68px;
    z-index: 1;
  }
}
