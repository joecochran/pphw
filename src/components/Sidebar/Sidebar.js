import React, { Component } from 'react';
import Draggable from '../Draggable/Draggable';
import './Sidebar.scss';
import Popup from '../Popup/Popup';

class Sidebar extends Component {
  render() {
    return (
      <div className="c-sidebar">
        <div className="c-sidebar__onboarding">
          <Popup title="foo" content="bar" action="baz" step={this.props.step} handleStepUpdate={this.props.handleStepUpdate} />
        </div>
        <Draggable name="Add lane" handleDrag={this.props.handleLaneDrag} handleDragEnd={this.props.handleDragEnd} />
        <Draggable name="Add bar" handleDrag={this.props.handleBarDrag} handleDragEnd={this.props.handleDragEnd} />
      </div>
    )
  }
  
}

export default Sidebar;
