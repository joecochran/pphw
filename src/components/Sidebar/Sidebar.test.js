import React from 'react';
import Sidebar from './Sidebar';
import { shallow } from 'enzyme';

describe('<Sidebar>', () => {
  const wrapper = shallow(<Sidebar />);

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
