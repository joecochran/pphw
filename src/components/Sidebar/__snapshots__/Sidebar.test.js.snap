// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar> matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="c-sidebar__onboarding"
        >
          <Popup
            action="baz"
            content="bar"
            title="foo"
          />
        </div>,
        <Draggable
          name="Add lane"
        />,
        <Draggable
          name="Add bar"
        />,
      ],
      "className": "c-sidebar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Popup
            action="baz"
            content="bar"
            title="foo"
          />,
          "className": "c-sidebar__onboarding",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "action": "baz",
            "content": "bar",
            "handleStepUpdate": undefined,
            "step": undefined,
            "title": "foo",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleDrag": undefined,
          "handleDragEnd": undefined,
          "name": "Add lane",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleDrag": undefined,
          "handleDragEnd": undefined,
          "name": "Add bar",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="c-sidebar__onboarding"
          >
            <Popup
              action="baz"
              content="bar"
              title="foo"
            />
          </div>,
          <Draggable
            name="Add lane"
          />,
          <Draggable
            name="Add bar"
          />,
        ],
        "className": "c-sidebar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Popup
              action="baz"
              content="bar"
              title="foo"
            />,
            "className": "c-sidebar__onboarding",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "action": "baz",
              "content": "bar",
              "handleStepUpdate": undefined,
              "step": undefined,
              "title": "foo",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleDrag": undefined,
            "handleDragEnd": undefined,
            "name": "Add lane",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleDrag": undefined,
            "handleDragEnd": undefined,
            "name": "Add bar",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
